datafusion/sql/src/unparser/expr.rs

```diff
@@ -600,6 +600,52 @@ impl Unparser<'_> {
             }
         }
 
+        if func_name.to_lowercase() == "round" {
```

Comment:
I don't think this should be in DataFusion. This isn't a SQL dialect difference, its a difference in how the functions are defined between systems. I think this should be implemented as part of the AST rewriter functionality we have

Reply:
Hm, it seems it follows the same dialect pattern when we specify what cast data type should be used to unparse specific expression.

Reply:
Yeah, I'm not sure this is the right pattern we want. Its a couple of functions now, but this won't scale to support all of the functions that all of the databases we want to support could unparse to. Maybe we need something stronger than a "Dialect", something that knows about how functions work.

Reply:
Yeah, it looks like we can start from AST we have and once we have better understanding we will be able to contribute it back to DataFusion. I have concern that unparser produces 'invalid' ast that can't be used so intuition is telling me that improvements should go to unparser so everyone can benefit from this work but if we can think of this as temporary solution with the target goal of contributing this as unparser optimization and dialects will be able to expose custom analyzers than this sounds like a good plan. Note: we found out that after unparsing cost of some queries increased so we need an optimization layer anyway IMO.